### ISSUE TITLE
tensor.cuda() is slow after invoking model(input) and loss.backward()

### ISSUE THREAD

**Issue by DUANJINHAO** [pytorch/pytorch#18786]
### I'm trying to test the costing of time in the main procedures in train-step, in order to accelerate my training. Then I found that `tensor.cuda()` become slow after executing `inference` and `backward.`
Here are the codes and the results with only load batch from dataloader and convert it to cuda:

    for i, (b, target) in enumerate(train_loader):
        start = time.time()
        target_cuda = target.cuda()
        input = b.cuda().half()
        now = time.time()
        cost = now - start
        print(cost)
        # output = model(input)
        # loss = criterion(output[0], target_cuda)
        # loss.backward()`
  
Results:
0.02122783660888672
0.06698179244995117
0.02122044563293457
0.02604532241821289
0.020719528198242188
0.02075052261352539
0.02080059051513672
0.020093202590942383
0.0195310115814209
0.0453181266784668
0.03698134422302246
0.03193855285644531
0.020868778228759766
0.03483223915100098
0.0204164981842041
It's really fast.
And after I cancel the comments in inference and backward. 
Result: with `model(input)` and `loss.backward()`
0.020980358123779297
0.27107763290405273
0.10506486892700195
0.668220043182373
0.6554262638092041
0.6526932716369629
0.6525142192840576
0.6601922512054443
0.6615090370178223
0.6599345207214355
0.6484317779541016
0.6491255760192871
0.6507673263549805
0.651357889175415
It's really slow, and it takes the most of time in current train-step.
I also add `torch.cuda.synchronize()`  following the guide in [here](https://discuss.pytorch.org/t/cuda-extremely-slow-after-calling-loss-backward/34371). It do make `.cuda()` faster but slow `backward().`
- [Discussion Forum](https://discuss.pytorch.org/)

**Comment by DUANJINHAO** [comment]
After adding `torch.cuda.synchronize()` behind `model(input)` and `loss.backward()`. I can see the real cost of time. So if where is a way to accelerate `loss.backward()`? Does tensor core can do that?

**Comment by ezyang** [comment]
The issue tracker is not an appropriate venue to debug slowness in your model. I recommend posting your full model on the forums and letting them help you. If you find there is some specific operation in PyTorch which ought to be faster, file a bug for it.
